dotnet/src/Connectors/Connectors.AI.OpenAI/TextToImage/AzureOpenAITextToImageService.cs

```diff
@@ -3,209 +3,161 @@
 
-        return result.Id;
+        return imageGenerations.Value.Data[0].Url.AbsoluteUri;
     }
 
-    /// <summary>
-    /// Retrieve the results of an text to image operation.
-    /// </summary>
-    /// <param name="operationId">The operationId that identifies the original text to image request.</param>
-    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
-    /// <returns></returns>
-    private async Task<AzureOpenAITextToImageResponse> GetTextToImageResultAsync(string operationId, CancellationToken cancellationToken = default)
+    private static OpenAIClientOptions GetClientOptions(HttpClient? httpClient, int? maxRetryCount, string? apiVersion)
     {
-        var operationLocation = this.GetUri(GetImageOperation, operationId);
-
-        var retryCount = 0;
+        OpenAIClientOptions.ServiceVersion version = apiVersion switch
+        {
+            "2022-12-01" => OpenAIClientOptions.ServiceVersion.V2022_12_01,
+            "2023-05-15" => OpenAIClientOptions.ServiceVersion.V2023_05_15,
+            "2023-06-01-preview" => OpenAIClientOptions.ServiceVersion.V2023_06_01_Preview,
+            "2023-07-01-preview" => OpenAIClientOptions.ServiceVersion.V2023_07_01_Preview,
+            "2023-08-01-preview" => OpenAIClientOptions.ServiceVersion.V2023_08_01_Preview,
+            _ => OpenAIClientOptions.ServiceVersion.V2023_09_01_Preview
+        };
 
-        while (true)
+        maxRetryCount ??= 5;
+        var options = new OpenAIClientOptions(version)
         {
-            if (this._maxRetryCount == retryCount)
-            {
-                throw new KernelException("Reached maximum retry attempts");
-            }
-
-            using var response = await this._core.ExecuteRequestAsync(operationLocation, HttpMethod.Get, null, cancellationToken).ConfigureAwait(false);
-            var responseJson = await response.Content.ReadAsStringWithExceptionMappingAsync().ConfigureAwait(false);
-            var result = OpenAITextToImageClientCore.JsonDeserialize<AzureOpenAITextToImageResponse>(responseJson);
-
-            if (result.Status.Equals(AzureOpenAIImageOperationStatus.Succeeded, StringComparison.OrdinalIgnoreCase))
-            {
-                return result;
-            }
-            else if (this.IsFailedOrCancelled(result.Status))
-            {
-                throw new KernelException($"Azure OpenAI text to image {result.Status}");
-            }
-
-            if (response.Headers.TryGetValues("retry-after", out var afterValues) && long.TryParse(afterValues.FirstOrDefault(), out var after))
-            {
-                await Task.Delay(TimeSpan.FromSeconds(after), cancellationToken).ConfigureAwait(false);
-            }
-
-            // increase retry count
-            retryCount++;
-        }
-    }
+            RetryPolicy = new RetryPolicy(maxRetries: Math.Max(0, maxRetryCount.Value)),
```

Comment:
Thank you for refactoring the class. Since you have introduced the Azure AI SDK usage here, let's ensure that it's configured in the same way as in [the other places where it's used](https://github.com/microsoft/semantic-kernel/blob/e905aa9d3fc3e2f654f4f9a25729d063068b8004/dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientCore.cs#L479) to maintain consistent behavior across our AI services and connectors.